src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksDtoToPaymentWorksVendorConversionServiceImpl.java

```diff
@@ -197,21 +199,56 @@ private void extractCustomFields(PaymentWorksVendor stgNewVendor, PaymentWorksNe
     }
 
     private void processEachPaymentWorksCustomFieldReceived(PaymentWorksVendor stgNewVendor, PaymentWorksNewVendorRequestDetailDTO pmwNewVendorDetailDTO, List<String> customFieldErrors) {
-        Map<String, String> customFieldsDTOMap = createCustomFieldIdCustomFieldValueMapForReceivedPmwVendorData(pmwNewVendorDetailDTO.getCustom_fields());
-        for (String paymentWorksCustomFieldId : customFieldsDTOMap.keySet()) {
-            String customFieldValueFromPaymentWorks = customFieldsDTOMap.get(paymentWorksCustomFieldId);
-            PaymentWorksFieldMapping fieldMapping = findPaymentWorksFieldMapping(paymentWorksCustomFieldId);
+        if (paymentWorksFormModeService.shouldUseLegacyFormProcessingMode()) {
+            Map<String, String> customFieldsDTOMap = createCustomFieldIdCustomFieldValueMapForReceivedPmwVendorData(pmwNewVendorDetailDTO.getCustom_fields());
```

Comment:
Is it only the way the customFieldsDTOMap is created that changes? Would it make more sense to only update this method?

Reply:
I changed the logic flow, but it might be cleaner with this idea, let me try and see what happens